src/pybind/mgr/telemetry/module.py

```diff
@@ -1246,6 +1246,15 @@ def toggle_channel(self, action, channels: Optional[List[str]] = None) -> Tuple[
 
         return 0, msg, ''
 
+    def restore_default_opt_setting(self, opt_name) -> None:
+        for o in self.MODULE_OPTIONS:
+            if o['name'] == opt_name:
+                default_val = o.get('default', None)
+                self.set_module_option(opt_name, default_val)
```

Comment:
as a user I wouldn't expect all my settings to be removed when I turned off the module - no other module does this afaik.

are you thinking this would be helpful for specific settings?